src/Lib2.hs

```diff
@@ -16,6 +16,12 @@ type Database = [(TableName, DataFrame)]
 
 -- Keep the type, modify constructors
 data ParsedStatement = ParsedStatement
```

Comment:
I think something like this might be more expressive:

```
data RelationalOperator = EQ | NE | LT | GT | LE | GE

type ColumnName = String

data WhereCriteria = ColumnName RelationalOperator Value

data AggregateFunction = Min | Sum

data Aggregate = AggregateFunction Column

type TableName = String

data ParsedStatement = ParsedStatement
{
    isShowTables :: Bool,
    aggregates :: [Aggregate],
    tables :: [TableName],
    whereClause :: [WhereCriteria]
} deriving (Show)
```